src/charm.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+"""Subordinate charm for TimescaleDB."""
+import os
+from subprocess import PIPE, Popen, check_call, check_output
+
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+
+class TimescaleDB(CharmBase):
+    """Subordinate charm for TimescaleDB."""
+
+    state = StoredState
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.upgrade_charm, self._on_upgrade_charm)
+
+    def _on_install(self, event):
+        if not hasattr(self.state, "installed"):
+            self.state.installed = False
+        elif self.state.installed:
+            return
+        try:
+            # test if postgresql installed yet
+            if not os.path.exists("/var/lib/postgresql"):
+                event.framework.model.unit.status = WaitingStatus(
+                    "waiting for postgresql to be installed"
+                )
+                event.defer()
+                return
+
+            # install dependencies
+            check_call(
+                ["sudo", "apt-get", "install", "-y", "apt-transport-https", "lsb-release", "wget"]
+            )
+
+            # add apt repo to sources
+            release = check_output(["lsb_release", "-c", "-s"]).decode("utf-8").rstrip("\n")
+            ps = Popen(
+                [
+                    "echo",
+                    "deb https://packagecloud.io/timescale/timescaledb/ubuntu/ {} main".format(
+                        release
+                    ),
```

Comment:
if we are targeting >Python3.6, we could f-strings for this.
```suggestion
                    "echo",
                    f"deb https://packagecloud.io/timescale/timescaledb/ubuntu/ {release} main"
```